pandas/core/series.py

```diff
@@ -2809,6 +2810,22 @@ def _try_cast(arr, take_fast_path):
             subarr = maybe_cast_to_datetime(arr, dtype)
             if not is_extension_type(subarr):
                 subarr = np.array(subarr, dtype=dtype, copy=copy)
+
+                d2 = data
+                if not is_list_like(d2):
+                    d2 = [data]
+                # Raises if coercion from unsigned to signed with neg data
+                if is_unsigned_integer_dtype(dtype) and any(x < 0 for x in d2):
```

Comment:
this is python any and will be non-performant, use what I gave before

``(d2<0).any()``